images/image.go

```diff
@@ -387,6 +382,49 @@ func Children(ctx context.Context, provider content.Provider, desc ocispec.Descr
 	return descs, nil
 }
 
+const readValidatedManifestBlobRetries = 3
+const readValidatedManifestBlobDelaySeconds = 5
+
+func readValidatedManifestBlobWithRetry(ctx context.Context, provider content.Provider, desc ocispec.Descriptor) ([]byte, error) {
+
+	retriesRemaining := readValidatedManifestBlobRetries
+
+	for {
+
+		blob, err := readValidatedManifestBlob(ctx, provider, desc)
+		if err == nil {
+			return blob, nil
+		}
+
+		if retriesRemaining > 0 {
+			log.G(ctx).
+				WithField("retries-remaining", retriesRemaining).
+				WithField("retry-delay-seconds", readValidatedManifestBlobDelaySeconds).
+				WithError(err).
+				Warn("readValidatedManifestBlob failed; retrying")
+
+			retriesRemaining -= 1
+
+			time.Sleep(time.Duration(readValidatedManifestBlobDelaySeconds) * time.Second)
+		} else {
+			return nil, err
+		}
+
+	}
+}
+
+func readValidatedManifestBlob(ctx context.Context, provider content.Provider, desc ocispec.Descriptor) ([]byte, error) {
+	p, err := content.ReadBlob(ctx, provider, desc)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := validateMediaType(ctx, p, desc.MediaType); err != nil {
+		return nil, fmt.Errorf("children: invalid desc %s: %w", desc.Digest, err)
```

Comment:
This is the error we've been seeing, with a JSON error in `err`, which suggests that the `json.Unmarshal` call in `validateMediaType` is where we're failing.  That in turn suggests we need to retry the ReadBlob _and_ the validateMediaType, which is why we extract `readValidatedManifestBlob` and wrap it in `readValidatedManifestBlobWithRetry`.